### PR TITLE
TensorFlow implementation of LowerTriangular Transform.

### PR DESCRIPTION
**Pull request by alexggmatthews**
Implements #439 . See comments in that Issue for discussion.

### CODE REVIEW

**Comment by alexggmatthews** [comment]
I've pre-squashed this one for you.

**Comment by alexggmatthews** [comment]
I'm running a timing experiment now.

**Comment by jameshensman** [comment]
@alexggmatthews any results on the timing?

File: GPflow/transforms.py [hunk 1]
```diff
@@ -301,7 +301,7 @@ class LowerTriangular(Transform):
     (N x N x D).
     """
 
-    def __init__(self, num_matrices=1, squeeze=False):
+    def __init__(self, N, num_matrices=1, squeeze=False):
         """
         Create an instance of LowerTriangular transform.
         Args:
```
Comment (jameshensman) [inline GPflow/transforms.py]: please add N to the args in the docstring

**Comment by jameshensman** [comment]
@alexggmatthews any results on the timing? 

PR looks good. I think it requires a version bump since you've changed the way that the triangular transform works.

**Comment by alexggmatthews** [comment]
I ran the mnist experiment of float32 py3.6 and tf1.2 on 1080 it is slightly faster with the new code. Making your doc and version changes now.

**Comment by alexggmatthews** [comment]
Ok done. Suggest we don't squash merge this as the commit comments may be useful.

File: GPflow/tf_wraps.py [hunk 2]
```diff
@@ -24,32 +24,31 @@
...
-    if M is None:
-        k = None
-    else:
-        k = int((M * 8 + 1) ** 0.5 / 2.0 - 0.5)
-    shape = tf.TensorShape([in_shape[0], k, k])
-    return [shape]
+def vec_to_tri( vectors, N ):
+	#Takes a D x M tensor `vectors'
+	#and maps it to a D x matrix_size X matrix_sizetensor
+	#where the where the lower
```
Comment (icouckuy) [inline GPflow/tf_wraps.py]: where the lower...

**Pull request merged**